metadeploy/api/models.py

```diff
@@ -904,6 +892,18 @@ def run(self, ctx, plan, steps, org):
         flow_coordinator.run(org)
 
 
+class ScratchOrgQuerySet(models.QuerySet):
+    def get_from_session(self, session):
+        """
+        Retrieve a ScratchOrg from the session by its ID.
+        The ID is placed in the session by GetScratchOrgIdFromQueryStringMiddleware.
+        """
+        scratch_org_id = session.get("scratch_org_id", None)
+        return self.filter(
+            uuid=scratch_org_id, status=self.model.Status.complete
+        ).first()
```

Comment:
Because `uuid` is a nullable field, do we want to also make sure that there was a `scratch_org_id` in the session at all? I'm not sure why we made that a nullable field...

```suggestion
        return (
            scratch_org_id
            and self.filter(
                uuid=scratch_org_id, status=self.model.Status.complete
            ).first()
            or None
        )
```

Reply:
Oh yeah, I remember wondering about that. Wouldn't it be better to make it non-nullable then? Also, what's the difference between `ScratchOrg.uuid` and `ScratchOrg.org_id`?

Reply:
> Wouldn't it be better to make it non-nullable then?

Personally I don't see why not. The only place we're creating scratch orgs is from the `scratch_org_post` method on the `PlanViewSet` (https://github.com/oddbird/MetaDeploy/blob/run-job/metadeploy/api/views.py#L311-L312), and the uuid is set right there on creation. @davisagli Do you have an argument for leaving `uuid` as a nullable field?

> Also, what's the difference between `ScratchOrg.uuid` and `ScratchOrg.org_id`?

The `uuid` is a unique identifier that has no utility beyond providing persistent access to the org for anonymous users. The `org_id` is an attribute of the SF org itself.

Reply:
I'm not aware of a reason it needs to be nullable.

Reply:
@jerivas I'll leave it to you to make the remaining changes to this PR then (or respond to the other comments).

Reply:
I made the UUID autogenerated too, takes the responsibility away from the Plan view (or anyone else who would want to use it)